fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/identifiers/FedoraID.java

```diff
@@ -181,21 +189,23 @@ public String getMementoString() {
     }
 
     /**
-     * Add a part onto this identifier.
-     * @param addition The additional string.
+     * Create a new FedoraID using the resource ID as the base
+     * @param addition One or more additional strings.
      * @return new FedoraID for the new identifier.
      */
-    public FedoraID addToResourceId(final String addition) {
-        return FedoraID.create(this.getResourceId() + (this.getResourceId().endsWith("/") ? "" : "/") + addition);
+    public FedoraID addToResourceId(final String... addition) {
+        final var newId = idBuilder(this.getResourceId(), addition);
+        return new FedoraID(newId);
```

Comment:
Isn't this the same as `return FedoraID.create(this.getResourceId(), addition)`?

Reply:
Sure, I can do that